shippy-node-example/shippy-app/js/shippy-storage.js

```diff
@@ -0,0 +1,162 @@
+		var xhr = new XMLHttpRequest(),
+			fileReader = new FileReader();
+
+		xhr.open("GET", file, true);
+		xhr.responseType = "blob";
+
+		// Wait the file to be loaded so we can extract its content
+		xhr.addEventListener("load", function () {
+			if (xhr.status === 200) {
+				// onload needed since Google Chrome doesn't support addEventListener for FileReader
+				fileReader.onload = function (evt) {
+					var content = evt.target.result;
+					var mimeType = fileToMime(file);
+					var data = {mimeType: mimeType, content: content};
+
+					try {
+						sessionStorage.setItem(key, JSON.stringify(data));
+						Lib.log('File successfully added to session storage', key);
+					}
+					catch (e) {
+						Lib.log("Storage failed: ", e);
+					}
+				};
+				fileReader.readAsText(xhr.response);
+			}
+		}, false);
+		xhr.send();
+	}
+
+	// get one file from the session storage
+	function get(key) {
+		let file = sessionStorage.getItem(key);
+		if (file) {
+			return JSON.parse(file);
+		}
+		return null;
+	}
+
+	// Iterate and add files to the session storage based on functions with extract the data required to do so
+	function addFiles(fileList, getPath, getKey, validate) {
+		for (let file of fileList) {
+			if (validate(file)) {
+				let key = getKey(file);
+				let path = getPath(file);
+
+				addFile(path, key);
+			}
+		}
+	}
+
+	// add script files to the session storage
+	function addScripts() {
+		let scripts = document.getElementsByTagName('script');
+		let getPath = function (script) {
+			return script.src;
+		};
+		let getKey = function (script) {
+			return '/'.concat(script.src.replace(script.baseURI, ''));
+		};
+		let validate = function (script) {
+			return script.src !== "";
+		};
+
+		addFiles(scripts, getPath, getKey, validate);
+	}
+
+	// add style sheet files to the session storage
+	function addStyleSheets() {
+		var styleSheets = document.styleSheets;
+		let getPath = function (styleSheet) {
+			return styleSheet.href;
+		};
+		let getKey = function (styleSheet) {
+			return '/'.concat(styleSheet.href.replace(styleSheet.ownerNode.baseURI, ''));
+		};
+		let validate = function (styleSheet) {
+			return styleSheet.href;
+		};
+
+		addFiles(styleSheets, getPath, getKey, validate);
+	}
+
+	function addHtmls() {
+		let content = Shippy.internal.initialHtml();
+		// if this function is triggered before the internal initial content is loaded, it will keep rescheduling its execution to the next 1s
+		// this may cause one infinite loop or other problems, but at least during testing everything worked fine
+		if (content) {
+			let data = {mimeType: extToMimes.html, content: content};
+			sessionStorage.setItem('/', JSON.stringify(data));
+			Lib.log('File successfully added to session storage', '/');
+			sessionStorage.setItem('/index.html', JSON.stringify(data));
+			Lib.log('File successfully added to session storage', '/index.html');
+		} else {
+			Lib.log('Initial HTML yet not loaded');
+			setTimeout(addHtmls, 1000);
```

Comment:
Feels a little weird to call it recursively with a timeout. Why should it not have been set? If `init` is called before DOMReady? I think in that case the API is used in a wrong way and throwing an error would be the better option?

Reply:
I think that this was from a prior implementation where I had an issue loading the page. 
After moving the init method to the onload event, that never happened again
I'll add a throw error just in case

Reply:
fixed